fix: spawn disabled in scene placed and network prefab registration (#4093)
MidLevel/MLAPI: com.unity.netcode.gameobjects/Documentation~/components/core/networkobject.md

You can use [NetworkBehaviours](networkbehaviour.md) to add your own custom Netcode logic to the associated NetworkObject.

### What is a valid NetworkObject?

There are two categories of NetworkObjects:

> * Dynamically intantiated

Reviewer: ```suggestion
* Dynamically instantiated
```

* [In-scene placed](../../basics/scenemanagement/inscene-placed-networkobjects.md)

The following provides the validity requirements for both types.

#### Dynamically instantiated network prefabs.

Requirements

* It must be a valid [network prefab](./networkobject.md#network-prefabs) created within the editor.

> * It must be registered within a network prefab list that is assigned to your NetworkManager.

Reviewer: ```suggestion
#### Dynamically instantiated network prefabs

Dynamically instantiated network prefabs must:

* Be a valid [network prefab](./networkobject.md#network-prefabs) created within the Editor.
* Be registered in a network prefab list that's assigned to your NetworkManager.
```

#### In-scene placed

Requirements

* It can be a valid network prefab instance within a scene.

> * It can be a GameObject with a `NetworkObject` component created within the scene while in the editor.

Reviewer: ```suggestion
#### In-scene placed network prefabs

In-scene placed network prefabs must:

* Be a valid network prefab instance within a scene.
* Be a GameObject with a NetworkObject component created within the scene while in the Editor.
```

### What is an invalid NetworkObject?

> Any GameObjects that have `NetworkObject` components added to them during runtime is **not supported** and will result in the NetworkObject's `GlobalObjectIdHash` being zero which would eventually result in synchronization issues. In the event you make this mistake, a warning message will be logged and the NetworkObject will not get spawned.

Reviewer: ```suggestion
GameObjects that have NetworkObject components added to them during runtime are **not supported** and will result in the NetworkObject's `GlobalObjectIdHash` being zero, which causes synchronization issues. In the event you make this mistake, a warning message will be logged and the NetworkObject won't be spawned.
```
